src/core/models/outputs/output_extended.c

```diff
@@ -1,4 +1,300 @@
 // Copyright 2021 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
+#include <inttypes.h>
+
+#include "core/address.h"
 #include "core/models/outputs/output_extended.h"
+#include "uthash.h"
+#include "utlist.h"
+
+output_extended_t* output_extended_new(address_t* addr, uint64_t amount, native_tokens_t* tokens,
+                                       feat_blk_list_t* feat_blocks) {
+  if (addr == NULL) {
+    printf("[%s:%d] invalid parameters\n", __func__, __LINE__);
+    return NULL;
+  }
+
+  if (amount < 1000000) {
```

Comment:
please give an identifier for this constant, ex `#define MIN_DUST_ALLOWANCE 1000000`